kyo-os-lib/shared/src/main/scala/kyo/files.scala

```diff
@@ -1,139 +1,405 @@
+            else
+                toJPath.getParent().exists.map { parentExists =>
+                    if parentExists then IOs(Files.write(toJPath, value.asJava, StandardOpenOption.WRITE)).map(_ => ())
+                    else IOs.unit
+                }
+
+        /** Methods to read files into streams
+          */
+
+        /** Reads a file returning its contents as a Stream of Strings
+          */
+        def readStream(charset: Charset = java.nio.charset.StandardCharsets.UTF_8): Stream[Unit, String, Fibers] < Resources =
+            Resources.acquireRelease(FileChannel.open(toJPath))(ch => ch.close()).map { fileCh =>
+                for
+                    ch <- Channels.init[Chunk[String] | Stream.Done](16)
+                    _  <- readLoop(fileCh, ByteBuffer.allocate(2048), ch, charset)
+                yield Streams.initChannel(ch)
+            }
+
+        /** Reads a file returning its contents as a Stream of Lines
+          */
+        def readLinesStream(charset: Charset = java.nio.charset.StandardCharsets.UTF_8): Stream[Unit, String, Fibers] < Resources =
+            Resources.acquireRelease(FileChannel.open(toJPath))(ch => ch.close()).map { fileCh =>
+                Channels.init[Chunk[String] | Stream.Done](16).map { ch =>
+                    readLoop(fileCh, ByteBuffer.allocate(2048), ch, charset).map(_ => Streams.initChannel(ch))
+                }
+            }
+
+        /** Reads a file returning its contents as a Stream of Bytes
+          */
+        def readBytesStream: Stream[Unit, Byte, Fibers] < Resources =
+            Resources.acquireRelease(FileChannel.open(toJPath))(ch => ch.close()).map { fileCh =>
+                Channels.init[Chunk[Byte] | Stream.Done](16).map { ch =>
+                    readLoopBytes(fileCh, ByteBuffer.allocate(2048), ch).map(_ => Streams.initChannel(ch))
+                }
+            }
+
+        private def readLoop(
+            fileChannel: FileChannel,
+            buf: ByteBuffer,
+            channel: Channel[Chunk[String] | Stream.Done],
+            charset: Charset
+        ) =
+            for
+                bytesRead <- IOs(fileChannel.read(buf))
+                f = Loops.transform(bytesRead) { b =>
+                    if b == -1 then
+                        for
+                            _ <- channel.put(Stream.Done)
+                        yield Loops.done(())
+                    else
+                        for
+                            s <- IOs {
+                                val arr = new Array[Byte](b)
+                                buf.get(arr)
+                                new String(arr, charset)
+                            }
+                            _            <- channel.put(Chunks.init(s))
+                            newBytesRead <- IOs(fileChannel.read(buf))
+                        yield Loops.continue(newBytesRead)
+                }
+                _ <- IOs(f)
```

Comment:
There's no need for this `IOs` suspension given that the loop already suspends side effects.